java/java-quiz.md

```diff
@@ -871,9 +871,25 @@ public class Solution {
 
 - [ ] int pi = 3.141;
 - [ ] decimal pi = 3.141;
-- [ ] double pi = 3.141;
-- [x] float pi = 3.141;
-
+- [x] double pi = 3.141;
+- [ ] float pi = 3.141;
+
+**Reasoning:** 
+public class TestReal {
+    public static void main (String[] argv)
+    {
+      double pi = 3.14159265;       //accuaracy upto 15 digits
+      float pi2 = 3.141F;                //accuracy upto 6-7 digits
+  
+      System.out.println ("Pi=" + pi);
+      System.out.println ("Pi2" + pi2;
+    }
+  }
+  
+  //The default Java type which Java will be using for a float variable will be double. 
+  //So, even if you declare any variable as float, what the compiler has to actually do is to assign a double value to a float variable, 
+  //which is not possible. So, to tell the compiler to treat this value as a float, that 'F' is used.
```

Comment:
```suggestion
 ```The default Java type which Java will be using for a float variable will be double. So, even if you declare any variable as float, what the compiler has to actually do is to assign a double value to a float variable, which is not possible. So, to tell the compiler to treat this value as a float, that 'F' is used.
 ```
```
add "```" at the end